### ISSUE TITLE
Ubuntu 20.04 LTS doesn't package sip as PyQt5.sip

### ISSUE THREAD

**Issue by The-Compiler** [qutebrowser/qutebrowser#6082]
Since PyQt5 5.11, the [sip module is supposed to be imported](https://www.riverbankcomputing.com/static/Docs/PyQt5/incompatibilities.html#pyqt-v5-11) as `PyQt5.sip`.

Thus, bff1b2a7d063f9093a50c8ed3ed94b777735e5d7 (dropping support for PyQt5 5.11 and below) dropped the old `sip` name (in `qt.py`).

However, it looks like Ubuntu 20.04 LTS (and possibly Debian?) packages `sip` in a weird way so that it still gets exported as `sip` rather than `PyQt5.sip`:

[python3-pyqt5](https://packages.ubuntu.com/focal/python3-pyqt5) depends on `sip-py3api-12.7`, provided by [python3-sip](https://packages.ubuntu.com/focal/python3-sip) and that [only provides the `sip` module](https://packages.ubuntu.com/focal/amd64/python3-sip/filelist).

As a workaround for affected users, this might work (untested):

```
ln -s /usr/lib/python3/dist-packages/sip.cpython-38-x86_64-linux-gnu.so /usr/lib/python3/dist-packages/PyQt5/ 
```

**Comment by cadadr** [comment]
Apparently this affects Guix too. I got 2.0.1 to build with the config below. IDK if it's immediately useful to you but maybe it might help someone having the same issue (hope that's okay to post, can delete otherwise).

`qutebrowser201sip.patch`:

```diff
--- a/qutebrowser/qt.py	2021-01-30 03:20:36.623204794 +0300
+++ b/qutebrowser/qt.py	2021-01-30 03:22:04.068829873 +0300
@@ -20,4 +20,7 @@
 """Wrappers around Qt/PyQt code."""
 
 # pylint: disable=unused-import
-from PyQt5 import sip
+try:
+    from PyQt5 import sip
+except:
+    import sip # type: ignore[import, no-redef]
```

`guix-user.scm`:

```scheme
(define gk-qutebrowser
  (package
    (inherit qutebrowser)
    (version "2.0.1")
    (source
     (origin
      (method url-fetch)
      (uri (string-append "https://github.com/qutebrowser/"
                          "qutebrowser/releases/download/v"
                           version "/qutebrowser-" version
                           ".tar.gz"))
      (patches (list "qutebrowser201sip.patch"))
      (sha256
       (base32 "0hmj19bp5dihzpphxz77377yfmygj498am0h23kxg5m3y5hqv65a"))))
    (inputs
     (cons* (list "python-importlib-resources" python-importlib-resources)
            (alist-delete "python-pypeg2"
                          (package-inputs qutebrowser))))))
```

**Comment by The-Compiler** [comment]
@cadadr Yup, that's pretty close to [what I removed originally](https://github.com/qutebrowser/qutebrowser/commit/bff1b2a7d063f9093a50c8ed3ed94b777735e5d7#diff-32f7640fddf268aed690c94af0a60ab821de144c994b09fcce44c1354999f1bb) and will readd for v2.0.2.

**Comment by cadadr** [comment]
@The-Compiler yeah, I just manually reversed the patch. I think of sending a patch to Guix to include the 2.x releases too. What do you think? Shall I wait for v2.0.2?

**Comment by The-Compiler** [comment]
@cadadr You didn't, the original code said `except ImportError:` rather than a plain `except:` (which is usually discouraged).

I just pushed a fix for this, but the v2.0.2 release will only happen somewhen next week - so your call.
